Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1239,19 +1239,19 @@ ExceptionOr<Ref<ElementInternals>> HTMLElement::attachInternals()
     return ElementInternals::create(*this);
 }
 
-static ExceptionOr<void> checkPopoverValidity(HTMLElement& element, PopoverVisibilityState expectedState, Document* expectedDocument = nullptr)
+static ExceptionOr<bool> checkPopoverValidity(HTMLElement& element, PopoverVisibilityState expectedState, Document* expectedDocument = nullptr)
 {
     if (element.popoverState() == PopoverState::None)
         return Exception { NotSupportedError, "Element does not have the popover attribute"_s };
 
-    if (!element.isConnected())
-        return Exception { InvalidStateError, "Element is not connected"_s };
+    if (element.popoverData()->visibilityState() != expectedState)
+        return false;
 
     if (expectedDocument && element.document() != *expectedDocument)
         return Exception { InvalidStateError, "Invalid when the document changes while showing or hiding a popover element"_s };
 
-    if (element.popoverData()->visibilityState() != expectedState)
-        return Exception { InvalidStateError, "Element has unexpected visibility state"_s };
+    if (!element.isConnected())
+        return Exception { InvalidStateError, "Element is not connected"_s };
```

Comment:
Can you move the connected check before the document check to match the spec?

Reply:
Good catch, fixed.